src/http_utils.cpp

```diff
@@ -244,9 +244,21 @@ void get_ip_str(
 {
     if(sa)
     {
+        int addrlen = sizeof(sockaddr_in);
+        if (AF_INET6 == sa->sa_family)
+        {
+            addrlen = sizeof(sockaddr_in6);
+        }
+
         char to_ret[NI_MAXHOST];
-        getnameinfo(sa, sizeof (struct sockaddr), to_ret, NI_MAXHOST, NULL, 0, NI_NUMERICHOST);
-        result = to_ret;
+        if (0 == getnameinfo(sa, addrlen, to_ret, NI_MAXHOST, NULL, 0, NI_NUMERICHOST))
+        {
+            result = to_ret;
+        }
+        else
```

Comment:
The rest looks good but I am not sure why you are clearing the result here.

Reply:
I can remove the `else` statement, no prob.

Because it's how i code 😉 For this type of function, i prefer return a cleared object instead of do nothing about it. Users of this function have always  a coherent object (setted or cleared). It's just my point of view.

Tell me what do you want, remove or not?

Reply:
And you are probably right. For now, remove it though for consistency with the rest of the codebase.

I am anyhow disagreeing with my old self here and I am going to rewrite the signature of those methods to avoid passing in input references (let the return values be return values) so the clear shouldn't be necessary anymore (sorry... I saw this part of my old code while reviewing your CR and I shivered).

Reply:
Ok.

It's never easy to re-read our old code 😄